food-truck-frontend/src/pages/truck/edit/[truckId].tsx

```diff
@@ -1,140 +1,166 @@
-        // If an error, log to console and kick out
-        if (err.response) {
-          console.log('Got error response code');
-        } else if (err.request) {
-          console.log('Did not receive Truck response');
-        } else {
-          console.log(err);
-        }
-
-        // Cancel
-        Router.replace("/");
-      });
-  }
-
-  onSubmit = (formData: any, response: AxiosResponse) => {
-    Router.replace(`/truck/${this.state.id}`);
+  const onSubmit = (formData: any, response: AxiosResponse) => {
+    Router.replace(`/truck/${state.id}`);
   }
 
-  onFail = (formData: any, response: AxiosResponse) => {
-    this.setState({
-      ...this.state,
+  const onFail = (formData: any, response: AxiosError) => {
+    setState({
+      ...state,
       message: `Failed to update truck details: ${JSON.stringify(response)}`
     });
   }
 
-  deleteTruck = () => {
-    api.delete(`/truck/delete/${this.state.id}`, {})
+  const deleteTruck = () => {
+    api.delete(`/truck/delete/${state.id}`, {})
       .then(res => Router.replace('/'))
       .catch(err => {
-        this.setState({
-          ...this.state,
+        setState({
+          ...state,
           message: `Failed to update truck details: ${JSON.stringify(err)}`
         })
       });
   }
+
+  if (state.userId == -1)
+    return <CircularProgress/>
+
+  return (
+    <Grid container alignItems="stretch">
+      <Grid container direction="row" alignItems="flex-start" className={classes.root}>
+        <Grid item className={classes.root}>
+          <Card>
+            <CardHeader title="Edit Truck Details"/>
+            <CardContent>
+              <Form submitMethod="PUT" submitUrl={'/truck/update'} onSuccessfulSubmit={onSubmit}
+                    onFailedSubmit={onFail}>
+                <TextField className={"hidden"} disabled label="Truck ID" name="truckId" defaultValue={state.id}/>
+                <TextField label="Truck Name" name="name" defaultValue={state.name}/>
+                <TextField label="Description" name="description"
+                           defaultValue={state.description}/>
+                <TextField label="Price Rating" name="priceRating"
+                           defaultValue={state.priceRating}/>
+                <MultiField
+                  title="Truck Tags"
+                  name="tags"
+                  variant="h6"
+                  value={state.tags}
+                />
+              </Form>
+            </CardContent>
+          </Card>
+        </Grid>
+        <Grid item direction="column" className={classes.root}>
+          <Card>
+            <CardHeader title="Upload Menu"/>
+            <CardContent>
+              <Grid container>
+                <Grid item>
+                  <form encType="multipart/form-data">
+                    <Button component="label">
+                      Upload Menu
+                      <input className="hidden" type="file" id="fileInput" name="file" onChange={onMenuChange}/>
```

Comment:
Given that the input is inside of the button, is there a way to click the button without firing the input event? I've seen bugs like that before, just don't know if it applies here.